dts/bindings/mtd/nordic,qspi-nor.yaml

```diff
@@ -52,6 +52,12 @@ properties:
       Specify the number of data lines and opcode used for writing.
       If not provided pp will be selected.
 
+  size-in-bytes:
+    type: boolean
+    required: false
+    description: |
+      Set to indicate that the size is specified in bytes not bits.
```

Comment:
Question: why not just make "size-in-bytes" property that would take the size value and have higher precedence than "size"/
I also find the name misleading because it kinda asks for "size-in-bytes=something"; the "size-is-in-bytes" would be more in-line, imho.

Reply:
I agree with both of your points, so there are two alternatives: 1 - make `size-in-bytes` take the size and have higher precedence or 2 - change the name to `size-is-in-bytes`.

One problem with 1 is that we cannot enforce (to my knowledge) that setting one of two properties is mandatory, hence both `size` and `size-in-bytes` must be `optional`, which can lead to incorrect configurations being accepted.

Reply:
I would then introduce optional size-in-bytes integer property and note to `size` that its value is ignored when the 1st is declared.

Reply:
Can we just fail compilation if the option is missing? Actually the size could be optional because, although we do not have support for it now, it could be read from device id data (READ ID (9Eh/9Fh) on some devices.

Reply:
I added a check in the driver, please check now.